clouddq/integration/bigquery/bigquery_client.py

```diff
@@ -208,3 +208,61 @@ def execute_query(
         )
 
         return query_job
+
+    def get_table_schema(self, table: str, project_id: str = None) -> dict:
+
+        client = self.get_connection(project_id=project_id)
+        try:
+            table_ref = client.get_table(table)
+        except KeyError as error:
+            raise KeyError(f"\n\nInput table `{table}` is not valid.\n{error}")
+        columns = {}
+        for column in table_ref.schema:
+            column_configs = {
+                "name": column.name,
+                "type": column.field_type,
+                "mode": column.mode,
+                "data_type": column.field_type,
+            }
+            columns[column.name.upper()] = column_configs
+        table_type = table_ref.table_type
+        logger.debug(f"Table Type is : {table_type}")
+        table_partitioning_type = table_ref.partitioning_type
+        logger.debug(f"Table Partitioning Type is : {table_partitioning_type}")
+        if table_partitioning_type:
+            partition_fields = []
+            time_partitioning = table_ref.time_partitioning
+            if time_partitioning:
+                partition_field = {}
+                time_partitioning_type = time_partitioning.type_
+                time_partitioning_field = time_partitioning.field
+                if not time_partitioning_field:
+                    partition_fields = "_PARTITIONTIME"
```

Comment:
How is `partition_fields` used? I wonder how we distinguish it being a string (i.e. "_PARTITIONTIME") from it being an array.

Reply:
_PARTITIONTIME is  a pseudocolumn for ingestion-time partitioning. The value of this column is the ingestion time for each row, truncated to the partition boundary (such as hourly or daily). This will be the default partition field. In case of metadata for GCS entities, it may have multiple partition fields so a list is used. For custom hive partitioning.

Reply:
It is intuitive for a field to be either a STRING or an ARRAY. I would prefer if we keep it consistent as an ARRAY in the CloudDQ implementation.

Is there any reason why we cannot force '_PARTITIONTIME' into the ARRAY format? e.g.

    "partition_fields": [{"name": "_PARTITIONTIME",
                          "partitioning_type": "DAY", <- ensure you get the correct partition type of day, hourly, or monthly
                          "type": "TIMESTAMP"}]

Reply:
Could you please also add an integration test for _PARTITIONTIME time-partitioned table?